lib/dynamoid/persistence.rb

```diff
@@ -250,6 +239,36 @@ def undump_hash_value(val)
         end
       end
 
+      def format_datetime(value)
+        if Dynamoid::Config.convert_date_to_string
+          value.iso8601
+        else
+          value.to_time.to_f
+        end
+      end
+
+      def parse_datetime(value)
+        return value if value.is_a?(Date) || value.is_a?(DateTime) || value.is_a?(Time)
+
+        case Dynamoid::Config.application_timezone
+          when :utc
+            time_for_timezone(value, 'UTC')
+          when :local
+            time_for_timezone(value, Time.now.getlocal.zone)
```

Comment:
`ActiveSupport::TimeZone` works badly with decree time and may loose 1 hour. That's why we used `Time.at(value).to_datetime` to convert time.